pkg/resolvers/aggregateResolver.go

```diff
@@ -0,0 +1,48 @@
+package resolvers
+
+import (
+	"fmt"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/kubernetes/pkg/registry/rbac/validation"
+)
+
+// AggregateRuleResolver conforms to the rbac/validation.AuthorizationRuleResolver interface and is used to aggregate multiple other AuthorizationRuleResolver into one resolver.
+type AggregateRuleResolver struct {
+	resolvers []validation.AuthorizationRuleResolver
+}
+
+// NewAggregateRuleResolver creates a new AggregateRuleResolver that will combine the outputs of all resolvers provided.
+func NewAggregateRuleResolver(resolvers ...validation.AuthorizationRuleResolver) *AggregateRuleResolver {
+	return &AggregateRuleResolver{
+		resolvers: resolvers,
+	}
+}
+
+// GetRoleReferenceRules calls GetRoleReferenceRules on each resolver and returns all returned rules and errors.
+func (a *AggregateRuleResolver) GetRoleReferenceRules(roleRef rbacv1.RoleRef, namespace string) ([]rbacv1.PolicyRule, error) {
```

Comment:
I'm a bit uncomfortable with our AggregateResolver atttempting to use this interface on the internal resolvers. We don't implement this method for our other resolvers, and this could easily lead to a panic if someone wasn't aware of that, or if the way that k8s used this method differed from the current impelmentation.

If we are going to fully implement this interface I think we should implement the interface for the crtb/prtb one as well.